wasmcloud_host/lib/wasmcloud_host_web/live/components/define_link_component.ex

```diff
@@ -12,32 +12,61 @@ defmodule DefineLinkComponent do
 
   def handle_event(
         "define_link",
-        %{
-          "actor_id" => actor_id,
-          "contract_id" => contract_id,
-          "link_name" => link_name,
-          "provider_id" => provider_id,
-          "values" => values
-        },
+        linkdef,
         socket
       ) do
+    IO.inspect(linkdef)
+    actor_id = Map.get(linkdef, "actor_id")
+    provider_id = Map.get(linkdef, "provider_id")
+    contract_id = Map.get(linkdef, "contract_id")
+    link_name = Map.get(linkdef, "link_name")
+    values = Map.get(linkdef, "values")
+
+    IO.inspect(values)
+
     values_map =
-      values
-      |> String.split(",")
-      |> Enum.flat_map(fn s -> String.split(s, "=") end)
-      |> Enum.chunk_every(2)
-      |> Enum.map(fn [a, b] -> {a, b} end)
-      |> Map.new()
+      case values do
+        "" ->
```

Comment:
does `"" | nil ->` work?

Reply:
It does not, I tried `||` as well and elixir doesn't seem to allow the conditionals in the pattern matching statement. Elixir also interprets `|` as a list operation 😢